jerry-core/ecma/builtin-objects/ecma-builtin-promise.c

```diff
@@ -577,10 +577,10 @@ ecma_builtin_promise_race_or_all (ecma_value_t this_arg, /**< 'this' argument */
       || ecma_get_object_type (ecma_get_object_from_value (array)) != ECMA_OBJECT_TYPE_ARRAY)
   {
     ecma_raise_type_error (ECMA_ERR_MSG ("Second argument is not an array."));
+    ecma_value_t current_error_value = JERRY_CONTEXT (error_value);
     ret = ecma_builtin_promise_reject_abrupt (capability);
-    ecma_free_value (JERRY_CONTEXT (error_value));
+    ecma_free_value (current_error_value);
```

Comment:
Shouldn't it be the responsibility of `ecma_builtin_promise_reject_abrupt` to free the error then? Between lines 135/136, between the `ecma_op_function_call` and `ecma_free_value (reject);`, `ecma_free_value (reason);` could achieve the same.

Reply:
Honestly, I'am not sure about that who's responsibility to free the `JERRY_CONTEXT (error_value)`.

Both solutions (I mean your suggestion and the PR's content) have advantages/disadvantages.
Your suggestion:
 - Pros: The `error_value` is always freed.
 - Cons: The `error_value` is also freed even if there wasn't a raised error before.

 PR's content:
 - Pros: Always the correct `error_value` is going to be freed because the developer raises and frees it.
 - Cons: The freeing part can be forgotten.

I do not insist to any of the proposed solutions but I'd ask @zherczeg and @LaszloLango about their opinions.

Reply:
> Cons: The error_value is also freed even if there wasn't a raised error before.

Note that `ecma_builtin_promise_reject_abrupt` is only called from here, so if it is called, then an error has been raised.

Reply:
Yes, it is called from one place. I would move the error construction / free there then.

Reply:
You succeeded in persuading me :) The PR is updated.